### PR TITLE
1376 make ontohub org database consistent

### LINKED ISSUE #1376: Make ontohub.org database consistent
ontohub.org has an inconsistent database:
The data migrations (on develop.ontohub.org, which has the same database content now) fail, because of invalid ontologies:

```
[...]
==  SetCountFieldsCorrectly: migrating ========================================
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:
Validation failed: Basepath can't be blank
[...]
/srv/http/ontohub/releases/20150605101629/db/data/20140823134219_set_count_fields_correctly.rb:9
[...]
```

This should not happen. Create a migration to fix such inconsistencies.

### PR DESCRIPTION
**Pull request by eugenk**
This shall fix #1376.

Actually, only the first commit (Set basepath where it is nil.	 671a2be) restores consistency. The other ones make sure that the data migrations pass. Since failing data migrations were the main reason of the issue, I put them into this PR as well.

Note:
The data migrations should be run manually with `RAILS_ENV=production nohup rake data:migrate &` to see the IDs of the failing OntologyVersions (see the third commit's message).

--

This way, 138 + 4350 OntologyVersions don't get a list of provers (timeout + other error).

You can find the file containing the logged failing IDs at `develop.ontohub.org:~ontohub/data_migration_provers_failures`

### CODE REVIEW

File: db/data/20141102203932_update_display_text_for_obo_sentences.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ def up
     obo_ontologies = Ontology.joins(:ontology_version).
       where(ontology_versions: {file_extension: '.obo'})
     obo_ontologies.each do |ontology|
-      ontology.sentences.find_each { |s| s.set_display_text! }
+      ontology.sentences.select([:id, :text]).find_each(&:set_display_text!)
```
Comment (0robustus1) [inline db/data/20141102203932_update_display_text_for_obo_sentences.rb]: How about a fancy-string: `%i(id text)`?

File: db/data/20150311194246_associate_provers_with_ontology_versions.rb [hunk 2]
```diff
@@ -9,4 +11,21 @@ def self.down
       ontology_version.save!
     end
   end
+
+  protected
+
+  def self.retry_provers_retrieval(ontology_version, max_attempts)
+    begin
```
Comment (0robustus1) [inline db/data/20150311194246_associate_provers_with_ontology_versions.rb]: the begin is unnecessary. You can just issue the `rescue`s on the method's-body.

File: db/data/20150311194246_associate_provers_with_ontology_versions.rb [hunk 3]
```diff
@@ -9,4 +11,21 @@ def self.down
...
+      ontology_version.retrieve_available_provers
+    rescue Net::ReadTimeout
+      if max_attempts > 0
+        puts "Timeout, trying again: OntologyVersion #{ontology_version.id}"
```
Comment (0robustus1) [inline db/data/20150311194246_associate_provers_with_ontology_versions.rb]: On STDERR?

File: db/data/20150311194246_associate_provers_with_ontology_versions.rb (hunk 2 above)
Comment (eugenk) [inline db/data/20150311194246_associate_provers_with_ontology_versions.rb]: Fixed with 62970f3.

File: db/data/20150311194246_associate_provers_with_ontology_versions.rb (hunk 3 above)
Comment (eugenk) [inline db/data/20150311194246_associate_provers_with_ontology_versions.rb]: Fixed with 62970f3.

File: db/data/20141102203932_update_display_text_for_obo_sentences.rb (hunk 1 above)
Comment (eugenk) [inline db/data/20141102203932_update_display_text_for_obo_sentences.rb]: Fixed with 7f10f65.

File: db/data/20140823134218_set_basepath_where_nil.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class SetBasepathWhereNil < ActiveRecord::Migration
+  def self.up
+    OntologyVersion.where(basepath: nil).find_each do |ov|
```
Comment (0robustus1) [inline db/data/20140823134218_set_basepath_where_nil.rb]: You could optimize this query by auto-loading the ontology (`includes`), and possibly by pre-selecting the fields.

**Comment by 0robustus1** [comment]
Apart from my last comment: :+1:

File: db/data/20140823134218_set_basepath_where_nil.rb (hunk 4 above)
Comment (eugenk) [inline db/data/20140823134218_set_basepath_where_nil.rb]: With `select` **and** `includes`, the columns don't get updated (no matter in what order I use those two methods). So I just use `includes` in 232f6a3.

**Comment by 0robustus1** [comment]
:+1:

**Pull request merged**